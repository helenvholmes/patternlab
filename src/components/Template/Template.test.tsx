import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

import {
  Template,
  TemplateBreakout,
  TemplateTop,
  TemplateMain,
  TemplateSidebar,
  TemplateBottom,
} from "./Template";
import Placeholder from "../Placeholder/Placeholder";

const breakout = <Placeholder variant="short">Breakout</Placeholder>;
const sidebar = "left";
const contentTop = <Placeholder>Content Top</Placeholder>;
const contentSidebar = <Placeholder>Left Sidebar</Placeholder>;
const contentMain = (
  <>
    <Placeholder>Main Content</Placeholder>
    <Placeholder variant="short">More Content</Placeholder>
  </>
);
const contentBottom = <Placeholder variant="short">Content Bottom</Placeholder>;

const templateChildren = (
  <Template sidebar={sidebar}>
    <TemplateBreakout>{breakout}</TemplateBreakout>
    <TemplateTop>{contentTop}</TemplateTop>
    <TemplateSidebar>{contentSidebar}</TemplateSidebar>
    <TemplateMain>{contentMain}</TemplateMain>
    <TemplateBottom>{contentBottom}</TemplateBottom>
  </Template>
);

describe("Template components accessibility", () => {
  it("passes axe accessibility test", async () => {
    const { container } = render(<>{templateChildren}</>);
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("Template components", () => {
  it("renders each section", () => {
    render(<>{templateChildren}</>);

    expect(screen.getByText("Breakout")).toBeInTheDocument();
    expect(screen.getByText("Content Top")).toBeInTheDocument();
    expect(screen.getByText("Left Sidebar")).toBeInTheDocument();
    expect(screen.getByText("Main Content")).toBeInTheDocument();
    expect(screen.getByText("More Content")).toBeInTheDocument();
    expect(screen.getByText("Content Bottom")).toBeInTheDocument();
  });

  it("renders a #mainContent id in the TemplateContent component", () => {
    const { container } = render(<>{templateChildren}</>);

    expect(container.querySelector("#mainContent")).toBeInTheDocument();
    expect(screen.getByRole("main")).toHaveAttribute("id", "mainContent");
  });

  it("renders the UI snapshot correctly", () => {
    const templateComponents = renderer
      .create(
        <Template sidebar={sidebar}>
          <TemplateBreakout>{breakout}</TemplateBreakout>
          <TemplateTop>{contentTop}</TemplateTop>
          <TemplateSidebar>{contentSidebar}</TemplateSidebar>
          <TemplateMain>{contentMain}</TemplateMain>
          <TemplateBottom>{contentBottom}</TemplateBottom>
        </Template>
      )
      .toJSON();

    expect(templateComponents).toMatchSnapshot();
  });
});
