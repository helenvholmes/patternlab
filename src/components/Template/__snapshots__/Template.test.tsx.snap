// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template components renders the UI snapshot correctly 1`] = `
<div
  className="css-0"
>
  <div
    className="css-fd96ed"
  >
    <div
      className="placeholder placeholder--short"
    >
      Breakout
    </div>
  </div>
  <div
    className="css-1q0thn5"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Content Top
    </div>
  </div>
  <div
    className="css-s7bsmx"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Left Sidebar
    </div>
  </div>
  <main
    className="css-n7iznt"
    id="mainContent"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Main Content
    </div>
    <div
      className="placeholder placeholder--short"
    >
      More Content
    </div>
  </main>
  <div
    className="css-1yhc1wu"
  >
    <div
      className="placeholder placeholder--short"
    >
      Content Bottom
    </div>
  </div>
</div>
`;

exports[`Template components renders the UI snapshot correctly 2`] = `
<div
  className="css-0"
>
  <div
    className="css-fd96ed"
  >
    <div
      className="placeholder placeholder--short"
    >
      Breakout
    </div>
  </div>
  <div
    className="css-1q0thn5"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Content Top
    </div>
  </div>
  <main
    className="css-n7iznt"
    id="mainContent"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Main Content
    </div>
    <div
      className="placeholder placeholder--short"
    >
      More Content
    </div>
  </main>
  <div
    className="css-s7bsmx"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Right Sidebar
    </div>
  </div>
  <div
    className="css-1yhc1wu"
  >
    <div
      className="placeholder placeholder--short"
    >
      Content Bottom
    </div>
  </div>
</div>
`;

exports[`Template components renders the UI snapshot correctly 3`] = `
<div
  className="css-0"
>
  <div
    className="css-fd96ed"
  >
    <div
      className="placeholder placeholder--short"
    >
      Breakout
    </div>
  </div>
  <div
    className="css-1q0thn5"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Content Top
    </div>
  </div>
  <main
    className="css-n7iznt"
    id="mainContent"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Main Content
    </div>
    <div
      className="placeholder placeholder--short"
    >
      More Content
    </div>
  </main>
  <div
    className="css-1yhc1wu"
  >
    <div
      className="placeholder placeholder--short"
    >
      Content Bottom
    </div>
  </div>
</div>
`;

exports[`Template components renders the UI snapshot correctly 4`] = `
<div
  className="css-0"
>
  <div
    className="css-fd96ed"
  >
    <div
      className="placeholder placeholder--short"
    >
      Breakout
    </div>
  </div>
  <div
    className="css-1q0thn5"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Content Top
    </div>
  </div>
  <main
    className="css-n7iznt"
    id="mainContent"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Main Content
    </div>
    <div
      className="placeholder placeholder--short"
    >
      More Content
    </div>
  </main>
  <div
    className="css-1yhc1wu"
  >
    <div
      className="placeholder placeholder--short"
    >
      Content Bottom
    </div>
  </div>
</div>
`;

exports[`Template components renders the UI snapshot correctly 5`] = `
<div
  className="css-0"
>
  <main
    className="css-n7iznt"
    id="mainContent"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Main Content
    </div>
    <div
      className="placeholder placeholder--short"
    >
      More Content
    </div>
  </main>
</div>
`;

exports[`Template components renders the UI snapshot correctly 6`] = `
<div
  className="css-0"
>
  <main
    className="css-n7iznt"
    id="mainContent"
  >
    <div
      className="placeholder placeholder--undefined"
    >
      Main Content
    </div>
    <div
      className="placeholder placeholder--short"
    >
      More Content
    </div>
  </main>
</div>
`;
